bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -167,23 +193,32 @@ public boolean isScreenOn() throws InterruptedException, AndroidDebugBridgeDevic
 
     public Optional<Boolean> isHDMIOn() throws InterruptedException, AndroidDebugBridgeDeviceException,
             AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
+        if (channelFallbackMap.get(HDMI_STATE_CHANNEL) == FallbackModes.LOGCAT) {
+            return isHDMIOnWithLogcat();
+        }
         String result = runAdbShell("cat", "/sys/devices/virtual/switch/hdmi/state");
         if (result.equals("0") || result.equals("1")) {
             return Optional.of(result.equals("1"));
         } else {
-            String fallback = runAdbShell("logcat", "-d", "|", "grep", "hdmi", "|", "grep", "SWITCH_STATE=", "|",
-                    "tail", "-1");
-            if (fallback.contains("SWITCH_STATE=")) {
-                return Optional.of(fallback.contains("SWITCH_STATE=1"));
-            } else if (fallback.isEmpty()) {
-                // IF THE DEVICE DO NOT SUPPORT THIS VALUE IN LOGCAT THE USER WILL NEVER KNOW THE CHANNEL WON'T WORK
-                // FIND A BETTER SOLUTION
-                return Optional.empty();
-            }
-            throw new AndroidDebugBridgeDeviceReadException(HDMI_STATE_CHANNEL, result, fallback);
+            channelFallbackMap.put(HDMI_STATE_CHANNEL, FallbackModes.LOGCAT);
+            return isHDMIOnWithLogcat();
         }
     }
 
+    private Optional<Boolean> isHDMIOnWithLogcat() throws InterruptedException, AndroidDebugBridgeDeviceException,
+            AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
+        String result = runAdbShell("logcat", "-d", "|", "grep", "hdmi", "|", "grep", "SWITCH_STATE=", "|", "tail",
+                "-1");
+        if (result.contains("SWITCH_STATE=")) {
+            return Optional.of(result.contains("SWITCH_STATE=1"));
+        } else if (result.isEmpty()) {
+            // IF THE DEVICE DO NOT SUPPORT THIS VALUE IN LOGCAT THE USER WILL NEVER KNOW THE CHANNEL WON'T WORK
+            // FIND A BETTER SOLUTION
+            return Optional.empty();
+        }
+        throw new AndroidDebugBridgeDeviceReadException(HDMI_STATE_CHANNEL, result);
```

Comment:
```suggestion
        channelFallbackMap.remove(HDMI_STATE_CHANNEL);
        throw new AndroidDebugBridgeDeviceReadException(HDMI_STATE_CHANNEL, result);
```

Reply:
To the case it never tries the first method again and the fallback is not supported by the device.